src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -1809,6 +1809,11 @@ namespace NServiceBus.Pipeline
         public void Register<TRegisterStep>()
             where TRegisterStep : NServiceBus.Pipeline.RegisterStep, new () { }
         public void Register(NServiceBus.Pipeline.RegisterStep registration) { }
```

Comment:
Seeing this makes me wonder why not using `Add`, `AddOrUpdate`, `TryRemove`, `TryUpdate`

which would be more aligned with methods from ConcurrentDictionary https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=netcore-3.1

Reply:
That would imply renaming our current `Register` to `Add`, breaking user-code unnecessarily. The initial name I added was `AddOrReplace` and I renamed it to `RegisterOrReplace` to remove the discrepancy in the added API, instead of breaking the existing one.

Reply:
True about breaking, but I'm not expecting our users to have lots of lines of pipeline configuration. Also, that is the beauty of our obsolete practice that the compiler errors give guidance to use a different method and the fix is super simple.